### PR TITLE
fix reset memory issue

### LINKED ISSUE #2123: [BUG] crewai reset-memories -a throws an error
### Description

Looks like reset-memories is throwing an error on -a 

It worked on .100 but gives the following on .102 

root@1b0b0f74e1a9:/crewai/dev/ceo-cf/research_chat# crewai reset-memories -a
An unexpected error occurred: No crew found.
root@1b0b0f74e1a9:/crewai/dev/ceo-cf/research_chat# 

### Steps to Reproduce

Go to root folder and run 'crewai reset-memories -a'

### Expected behavior

Expected behaviour is that it clears the memory for all 

### Screenshots/Code snippets

Image

### Operating System

Other (specify in additional context)

### Python Version

3.10

### crewAI Version

0.102

### crewAI Tools Version

Assume 0.102

### Virtual Environment

Venv

### Evidence

Running in a Docker environment on MacOS 15.3.1

### Possible Solution

Implement memory clear that was working 

### Additional context

Small thing.. would like it to all deleted the /db as I need to manually delete it sometimes

### PR DESCRIPTION
**Pull request by bhancockio**
Closes #2123 

https://www.loom.com/share/a60dda3aa8e64e84b47e016143ad7e84?sid=0dba0c4d-f324-4e9a-bef2-874101499e53

### CODE REVIEW

**Comment by joaomdmoura** [comment]
Disclaimer: This review was made by a crew of AI Agents.

### Code Review Comment for PR #2182 - Memory Reset Fix

#### Overview
This pull request introduces changes to the `_reset_all_memories` method within `crew.py`, enhancing the way memory system attributes are accessed by using `getattr()`. This approach mitigates potential attribute access errors when memory systems have not been initialized, thus improving the robustness of the code.

#### Modified Code Analysis
The primary changes are as follows:

**Original Code:**
```python
memory_systems = [
    ("short term", self._short_term_memory),
    ("entity", self._entity_memory),
    ("long term", self._long_term_memory),
    ("task output", self._task_output_handler),
    ("knowledge", self.knowledge),
]
```

**Modified Code:**
```python
memory_systems = [
    ("short term", getattr(self, "_short_term_memory", None)),
    ("entity", getattr(self, "_entity_memory", None)),
    ("long term", getattr(self, "_long_term_memory", None)),
    ("task output", getattr(self, "_task_output_handler", None)),
    ("knowledge", getattr(self, "knowledge", None)),
]
```

#### Positive Aspects
1. **Error Handling Improvement:** The use of `getattr()` with a default value prevents `AttributeError` exceptions when memory systems are not initialized, enhancing the stability of the function.
2. **Consistent Behavior:** This modification ensures uniform access behavior across different memory systems, reducing the risk of inconsistent states.
3. **Adherence to Python Best Practices:** The change aligns with the EAFP (Easier to Ask for Forgiveness than Permission) principle, promoting a common idiom in Python programming.

#### Suggestions for Improvement
1. **Add Type Hints:**
   Incorporating type hints enhances code readability and facilitates static type checking. For example:
   ```python
   def _reset_all_memories(self) -> None:
       """Reset all available memory systems."""
       memory_systems: List[Tuple[str, Optional[Any]]] = [
           ("short term", getattr(self, "_short_term_memory", None)),
           ("entity", getattr(self, "_entity_memory", None)),
           ("long term", getattr(self, "_long_term_memory", None)),
           ("task output", getattr(self, "_task_output_handler", None)),
           ("knowledge", getattr(self, "knowledge", None)),
       ]
   ```

2. **Implement Logging for Diagnostics:**
   Adding logging can provide insights into which memory systems were accessed or reset:
   ```python
   for name, system in memory_systems:
       if system is None:
           logger.debug(f"Memory system '{name}' not initialized")
       else:
           logger.debug(f"Resetting memory system: {name}")
   ```

3. **Extract Memory Systems Configuration:**
   Defining a configuration list for memory systems can simplify modifications and improve readability:
   ```python
   MEMORY_SYSTEMS_CONFIG = [
       ("short term", "_short_term_memory"),
       ("entity", "_entity_memory"),
       ("long term", "_long_term_memory"),
       ("task output", "_task_output_handler"),
       ("knowledge", "knowledge"),
   ]
   ```

#### Additional Recommendations
- **Documentation:** Enhance the docstring of the `_reset_all_memories` method to specify behavior when memory systems are not initialized and document expected types.
- **Testing:** Develop unit tests to cover scenarios where memory systems may be uninitialized, validate the reset behavior, and handle edge cases effectively.
- **Post-Reset Validation:** Consider implementing checks to validate the states of memory systems following a reset, thus ensuring complete reset functionality.

#### Conclusion
The changes made in this pull request significantly enhance the robustness of the memory resetting functionality by safely handling potentially uninitialized attributes. The recommended improvements aim to foster better maintainability, readability, and observability of the code. The patch is approved with the suggested enhancements for further consideration.

File: src/crewai/crew.py [hunk 1]
```diff
@@ -1278,11 +1278,11 @@ def reset_memories(self, command_type: str) -> None:
...
-            ("short term", self._short_term_memory),
-            ("entity", self._entity_memory),
-            ("long term", self._long_term_memory),
-            ("task output", self._task_output_handler),
-            ("knowledge", self.knowledge),
+            ("short term", getattr(self, "_short_term_memory", None)),
+            ("entity", getattr(self, "_entity_memory", None)),
+            ("long term", getattr(self, "_long_term_memory", None)),
+            ("task output", getattr(self, "_task_output_handler", None)),
+            ("knowledge", getattr(self, "knowledge", None)),
```
Comment (lorenzejay) [inline src/crewai/crew.py:1285]: great work and thanks for sharing the loom

**Review by lorenzejay — APPROVED** [review]
